core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -996,7 +999,11 @@ private Frame decorrelateInputWithValueGenerator(RelNode rel, Frame frame) {
       }
       // If all correlation variables are now satisfied, skip creating a value
       // generator.
-      if (map.size() == corVarList.size()) {
+      for (CorRef corVar : corVarList) {
+        temp = temp.concat(".").concat(String.valueOf(corVar.field));
```

Comment:
if we can replace this with stream 
temp = corVarList.stream()
          .map(corVar -> "." + corVar.field)
          .collect(Collectors.joining());

Reply:
Done